Libraries/Azure.psm1

```diff
@@ -2397,23 +2397,67 @@ function Add-AzureAccountFromSecretsFile {
+            $null = Add-AzureRmAccount -ServicePrincipal -Tenant $TenantID -Credential $mycred
+        } elseif ($AzureContextFilePath) {
+            Write-LogInfo "Authenticating Azure PS session using saved context file $AzureContextFilePath..."
+            $null = Import-AzureRmContext -Path $AzureContextFilePath
+        } else {
+            $ContextFiles = Get-ChildItem -Path $PWD -Recurse | `
+                Where-Object { $_.Name.EndsWith(".json") } | Select-String -Pattern "AzureCloud" | Select Path
+            $ContextFiles = $ContextFiles.Path | Get-Unique
+            if ($ContextFiles.Count -eq 0) {
+                Write-LogWarn "No Azure authentication methods were available in Secret File."
+                Write-LogWarn "No Azure authentication context files detected in $PWD."
+
+                Write-LogWarn "Checking if current Azure powershell session is already authenticated..."
+                $selectedSubscription = Select-AzureRmSubscription -SubscriptionId $XmlSecrets.secrets.SubscriptionID -ErrorAction SilentlyContinue
+                if ( $selectedSubscription.Subscription.Id -eq $XmlSecrets.secrets.SubscriptionID ) {
+                    $SubscriptionSelected = $true
+                    Write-LogWarn "Authenticated."
+                } else {
+                    Write-LogWarn "Unable to proceed with unauthenticated Azure powershell session."
+                    Write-LogWarn "Please use one of the following method to authenticate this session."
+                    Write-LogWarn "1. Provide service principal details in XML secrets file."
+                    Write-LogWarn "    a. SubscriptionServicePrincipalClientID"
+                    Write-LogWarn "    b. SubscriptionServicePrincipalTenantID"
+                    Write-LogWarn "    c. SubscriptionServicePrincipalKey"
+                    Write-LogWarn "2. Provide the path of authenticated context file in XML secrets file."
+                    Write-LogWarn "    a. AzureContextFilePath"
+                    Write-LogWarn "3. Copy authenticated context file in $PWD"
+                    Write-LogWarn "4. Authenticate this current session by running Connect-AzureRmAccount command, and then run LISAv2 again."
+                }
+            } elseif ($ContextFiles.Count -eq 1)  {
+                $CustomAzureContextFilePath = $ContextFiles
+                Write-LogInfo "Authenticating with context file $CustomAzureContextFilePath"
+                $null = Import-AzureRmContext -Path $CustomAzureContextFilePath
+            } else {
+                Write-LogWarn "$($ContextFiles.Count) Azure context files found in $pwd."
+                $Counter = 1
+                foreach ( $file in $ContextFiles ) {
+                    Write-LogWarn "$Counter. $file"
+                }
+                Write-LogWarn "Please remove unwanted context files. Expected context files: 1."
+            }
+        }
+        if (-not $SubscriptionSelected) {
```

Comment:
You did not initialize this variable value, this will falls into exception.

Reply:
@juhlee-microsoft  Fixed.